


<!DOCTYPE html>
<html id="htmlId">
<head>
  <meta http-equiv="Content-Type" content="text/html;charset=windows-1252"> 
  <title>Coverage Report > CategoriaControllerTest</title>
  <style type="text/css">
    @import "../../css/coverage.css";
    @import "../../css/idea.min.css";
  </style>
  <script type="text/javascript" src="../../js/highlight.min.js"></script>
  <script type="text/javascript" src="../../js/highlightjs-line-numbers.min.js"></script>
</head>

<body>
<div class="content">
<div class="breadCrumbs">
Current scope:     <a href="../../index.html">all classes</a>
    <span class="separator">|</span>
    <a href="../index.html">com.pagueibaratoapi.pagueibaratoapi.unit</a>
</div>

<h1>Coverage Summary for Class: CategoriaControllerTest (com.pagueibaratoapi.pagueibaratoapi.unit)</h1>

<table class="coverageStats">
<tr>
  <th class="name">Class</th>
<th class="coverageStat 
">
  Class, %
</th>
<th class="coverageStat 
">
  Method, %
</th>
<th class="coverageStat 
">
  Line, %
</th>
</tr>
<tr>
  <td class="name">CategoriaControllerTest</td>
<td class="coverageStat">
  <span class="percent">
    100%
  </span>
  <span class="absValue">
    (1/1)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    100%
  </span>
  <span class="absValue">
    (37/37)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    86,8%
  </span>
  <span class="absValue">
    (191/220)
  </span>
</td>
</tr>

</table>

<br/>
<br/>


<pre>
<code class="sourceCode" id="sourceCode">&nbsp;package com.pagueibaratoapi.pagueibaratoapi.unit;
&nbsp;
&nbsp;import static org.junit.Assert.assertTrue;
&nbsp;import static org.junit.jupiter.api.Assertions.assertEquals;
&nbsp;import static org.mockito.ArgumentMatchers.any;
&nbsp;import static org.mockito.ArgumentMatchers.anyInt;
&nbsp;import static org.mockito.ArgumentMatchers.anyString;
&nbsp;import static org.mockito.Mockito.doThrow;
&nbsp;import static org.mockito.Mockito.verify;
&nbsp;import static org.mockito.Mockito.when;
&nbsp;
&nbsp;import java.util.ArrayList;
&nbsp;import java.util.List;
&nbsp;import java.util.NoSuchElementException;
&nbsp;import java.util.Optional;
&nbsp;
&nbsp;import org.junit.Before;
&nbsp;import org.junit.Test;
&nbsp;import org.mockito.InjectMocks;
&nbsp;import org.mockito.Mock;
&nbsp;import org.mockito.MockitoAnnotations;
&nbsp;import org.springframework.dao.DataIntegrityViolationException;
&nbsp;import org.springframework.web.server.ResponseStatusException;
&nbsp;
&nbsp;import com.pagueibaratoapi.controllers.CategoriaController;
&nbsp;import com.pagueibaratoapi.models.requests.Categoria;
&nbsp;import com.pagueibaratoapi.models.responses.ResponseCategoria;
&nbsp;import com.pagueibaratoapi.pagueibaratoapi.PagueiBaratoApiApplicationTests;
&nbsp;import com.pagueibaratoapi.repository.CategoriaRepository;
&nbsp;
&nbsp;
&nbsp;
<b class="fc">&nbsp;public class CategoriaControllerTest extends PagueiBaratoApiApplicationTests {</b>
&nbsp;
&nbsp;    @InjectMocks
&nbsp;    private CategoriaController categoriaController;
&nbsp;
&nbsp;    @Mock
&nbsp;    private CategoriaRepository categoriaRepository;
&nbsp;
&nbsp;    @Mock
&nbsp;    private Optional&lt;Categoria&gt; optionalCategoriaMock;
&nbsp;
&nbsp;    @Mock
&nbsp;    private List&lt;ResponseCategoria&gt; responseCategoria;
&nbsp;
&nbsp;    private Categoria categoria;
&nbsp;
&nbsp;    private Categoria categoriaInvalida;
&nbsp;
&nbsp;    private List&lt;Categoria&gt; categorias;
&nbsp;
&nbsp;
&nbsp;
&nbsp;    @Before
&nbsp;    public void setUp() {
&nbsp;        // Inicializa os objetos mockados
<b class="fc">&nbsp;        MockitoAnnotations.openMocks(this);</b>
&nbsp;
&nbsp;        // Inicializa o objeto categoria, atribuindo valores aos seus atributos
<b class="fc">&nbsp;        this.inicializarCategoria();</b>
&nbsp;
&nbsp;        // Inicializa a lista de categorias, atribuindo valores aos seus atributos
<b class="fc">&nbsp;        this.inicializarCategorias();</b>
&nbsp;    }
&nbsp;
&nbsp;    private void inicializarCategoria() {
<b class="fc">&nbsp;        categoria = new Categoria();</b>
<b class="fc">&nbsp;        categoria.setNome(&quot;Eletrodomésticos&quot;);</b>
<b class="fc">&nbsp;        categoria.setDescricao(&quot;Eletrodomésticos em geral&quot;);</b>
&nbsp;
<b class="fc">&nbsp;        categoriaInvalida = new Categoria();</b>
<b class="fc">&nbsp;        categoriaInvalida.setNome(&quot;Eletrodomésticos em geral e mais um pouco&quot;);</b>
<b class="fc">&nbsp;        categoriaInvalida.setDescricao(&quot;Eletrodomésticos em geral e mais um pouco.&quot;);</b>
&nbsp;    }
&nbsp;
&nbsp;    private void inicializarCategorias() {
<b class="fc">&nbsp;        Categoria categoria1 = new Categoria();</b>
<b class="fc">&nbsp;        categoria1.setId(1);</b>
<b class="fc">&nbsp;        categoria1.setNome(&quot;Eletrodomésticos&quot;);</b>
<b class="fc">&nbsp;        categoria1.setDescricao(&quot;Eletrodomésticos em geral&quot;);</b>
&nbsp;
<b class="fc">&nbsp;        Categoria categoria2 = new Categoria();</b>
<b class="fc">&nbsp;        categoria2.setId(2);</b>
<b class="fc">&nbsp;        categoria2.setNome(&quot;Eletroeletrônicos&quot;);</b>
<b class="fc">&nbsp;        categoria2.setDescricao(&quot;Eletroeletrônicos em geral&quot;);</b>
&nbsp;
<b class="fc">&nbsp;        Categoria categoria3 = new Categoria();</b>
<b class="fc">&nbsp;        categoria3.setId(3);</b>
<b class="fc">&nbsp;        categoria3.setNome(&quot;Eletroportáteis&quot;);</b>
<b class="fc">&nbsp;        categoria3.setDescricao(&quot;Eletroportáteis em geral&quot;);</b>
&nbsp;
<b class="fc">&nbsp;        categorias = List.of(categoria1, categoria2, categoria3);</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;
&nbsp;    
&nbsp;    /* -------------------------  CRIAÇÃO DE CATEGORIA  ------------------------- */
&nbsp;
&nbsp;    @Test
&nbsp;    public void criarCategoriaComSucesso() throws Exception {
&nbsp;
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.existsByNomeIgnoreCase(any())).thenReturn(false);</b>
<b class="fc">&nbsp;        when(categoriaRepository.save(any())).thenReturn(categoria);</b>
&nbsp;
<b class="fc">&nbsp;        ResponseCategoria response = categoriaController.criar(categoria);</b>
&nbsp;
<b class="fc">&nbsp;        assertTrue(categoria.getNome().equals(response.getNome()));</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void criarCategoriaComNomeExistente() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.existsByNomeIgnoreCase(any())).thenReturn(true);</b>
&nbsp;
&nbsp;        try {
<b class="fc">&nbsp;            categoriaController.criar(categoria);</b>
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(409, e.getRawStatusCode());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void criarCategoriaComDadosInvalidos() throws Exception {
&nbsp;
&nbsp;        try {
<b class="fc">&nbsp;            categoriaController.criar(null);</b>
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(400, e.getRawStatusCode());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void criarCategoriaComExcecaoDataViolation() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.save(categoria)).thenThrow(new DataIntegrityViolationException(&quot;erro_insercao&quot;));</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.criar(categoria);</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(500, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;erro_insercao&quot;, e.getReason());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void criarCategoriaComExcecaoIllegalArgument() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.save(categoria)).thenThrow(new IllegalArgumentException(&quot;erro_inesperado&quot;));</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.criar(categoria);</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(500, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;erro_inesperado&quot;, e.getReason());</b>
<b class="fc">&nbsp;            assertTrue(e.getCause().toString().contains(&quot;java.lang.IllegalArgumentException&quot;));</b>
<b class="nc">&nbsp;        }</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void criarCategoriaComExcecao() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.save(any())).thenThrow(new RuntimeException(&quot;erro_inesperado&quot;));</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.criar(categoria);</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(500, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;erro_inesperado&quot;, e.getReason());</b>
<b class="fc">&nbsp;            assertTrue(e.getCause().toString().contains(&quot;RuntimeException&quot;));</b>
<b class="nc">&nbsp;        }</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    /* -------------------------------------------------------------------------- */
&nbsp;
&nbsp;
&nbsp;
&nbsp;
&nbsp;
&nbsp;    /* ---------------------  LEITURA DE CATEGORIA POR ID  ---------------------- */
&nbsp;
&nbsp;    @Test
&nbsp;    public void lerCategoriaPorIdComSucesso() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.findById(any())).thenReturn(Optional.ofNullable(categoria));</b>
&nbsp;
<b class="fc">&nbsp;        ResponseCategoria response = categoriaController.ler(1);</b>
&nbsp;
<b class="fc">&nbsp;        assertTrue(categoria.getNome().equals(response.getNome()));</b>
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void lerCategoriaPorIdComExcecaoNoSuchElement() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.findById(any())).thenThrow(new NoSuchElementException(&quot;nao_encontrado&quot;));</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.ler(1986);</b>
&nbsp;            
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(404, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;nao_encontrado&quot;, e.getReason());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void lerCategoriaPorIdComExcecao() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.findById(any())).thenReturn(optionalCategoriaMock);</b>
<b class="fc">&nbsp;        when(optionalCategoriaMock.get()).thenThrow(new NullPointerException());</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.ler(1986);</b>
&nbsp;            
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(500, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;erro_inesperado&quot;, e.getReason());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;    }
&nbsp;
&nbsp;    /* -------------------------------------------------------------------------- */
&nbsp;
&nbsp;
&nbsp;
&nbsp;
&nbsp;
&nbsp;    /* ------------------------  LISTAGEM DE CATEGORIAS  ------------------------ */
&nbsp;
&nbsp;    @Test
&nbsp;    public void listarCategoriasComSucesso() throws Exception {
&nbsp;
&nbsp;        // Configura o mock do repository para retornar a lista de categorias
<b class="fc">&nbsp;        when(categoriaRepository.findAll()).thenReturn(categorias);</b>
&nbsp;
&nbsp;        // Chama o método para listar categorias do controller
<b class="fc">&nbsp;        List&lt;ResponseCategoria&gt; categoriasResponse = categoriaController.listar();</b>
&nbsp;
&nbsp;        // Verifica se a lista de categorias retornada pelo controller é igual a lista de categorias criada
<b class="fc">&nbsp;        assertTrue(!categoriasResponse.isEmpty());</b>
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void listarCategoriasComSucessoSemRetorno() throws Exception {
&nbsp;
&nbsp;        // Cria uma lista de categorias vazia que o mock do repository deve retornar
<b class="fc">&nbsp;        List&lt;Categoria&gt; categoriasList = new ArrayList&lt;Categoria&gt;();</b>
&nbsp;
&nbsp;        // Configura o mock do repository para retornar a lista de categorias
<b class="fc">&nbsp;        when(categoriaRepository.findAll()).thenReturn(categoriasList);</b>
&nbsp;
&nbsp;        // Chama o método para listar categorias do controller
<b class="fc">&nbsp;        List&lt;ResponseCategoria&gt; categoriasResponse = categoriaController.listar();</b>
&nbsp;
&nbsp;        // Verifica se a lista de categorias retornada pelo controller é igual a lista de categorias criada
<b class="fc">&nbsp;        assertTrue(categoriasResponse.isEmpty());</b>
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void listarCategoriasComExcecaoNullPointer() throws Exception {
&nbsp;
&nbsp;        // Configura o mock do repository para que retorne uma exceção NullPointerException
<b class="fc">&nbsp;        when(categoriaRepository.findAll()).thenThrow(new NullPointerException());</b>
&nbsp;
&nbsp;        try {
&nbsp;
&nbsp;            // Chama o método para listar categorias do controller
<b class="fc">&nbsp;            categoriaController.listar();</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
&nbsp;            // Verifica se o código de resposta é 404
<b class="fc">&nbsp;            assertEquals(404, e.getRawStatusCode());</b>
&nbsp;            // Verifica se a mensagem de erro é &quot;nao_encontrado&quot;
<b class="fc">&nbsp;            assertEquals(&quot;nao_encontrado&quot;, e.getReason());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void listarCategoriasComExcecaoUnsupportedOperation() throws Exception {
&nbsp;
&nbsp;        // Configura o mock do repository para que retorne uma exceção UnsupportedOperationException
<b class="fc">&nbsp;        when(categoriaRepository.findAll()).thenThrow(new UnsupportedOperationException());</b>
&nbsp;
&nbsp;        try {
&nbsp;
&nbsp;            // Chama o método para listar categorias do controller
<b class="fc">&nbsp;            categoriaController.listar();</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
&nbsp;            // Verifica se o código de resposta é 404
<b class="fc">&nbsp;            assertEquals(500, e.getRawStatusCode());</b>
&nbsp;            // Verifica se a mensagem de erro é &quot;erro_inesperado&quot;
<b class="fc">&nbsp;            assertEquals(&quot;erro_inesperado&quot;, e.getReason());</b>
&nbsp;            // Verifica se a causa da exceção é UnsupportedOperationException
<b class="fc">&nbsp;            assertTrue(e.getCause().toString().contains(&quot;java.lang.UnsupportedOperationException&quot;));</b>
<b class="nc">&nbsp;        }</b>
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void listarCategoriasComExcecao() throws Exception {
&nbsp;
&nbsp;        // Configura o mock do repository para que retorne uma exceção
<b class="fc">&nbsp;        when(responseCategoria.add(any())).thenThrow(new NullPointerException());</b>
&nbsp;
&nbsp;        try {
&nbsp;
&nbsp;            // Chama o método para listar categorias do controller
<b class="fc">&nbsp;            categoriaController.listar();</b>
&nbsp;
&nbsp;        }
<b class="nc">&nbsp;        catch (ResponseStatusException e) {</b>
&nbsp;            // Verifica se o código de resposta é 404
<b class="nc">&nbsp;            assertEquals(500, e.getRawStatusCode());</b>
&nbsp;            // Verifica se a mensagem de erro é &quot;erro_inesperado&quot;
<b class="nc">&nbsp;            assertEquals(&quot;erro_inesperado&quot;, e.getReason());</b>
&nbsp;            // Verifica se a causa da exceção é UnsupportedOperationException
<b class="nc">&nbsp;            assertTrue(e.getCause().toString().contains(&quot;java.lang.UnsupportedOperationException&quot;));</b>
<b class="fc">&nbsp;        }</b>
&nbsp;    }
&nbsp;
&nbsp;    /* -------------------------------------------------------------------------- */
&nbsp;
&nbsp;
&nbsp;
&nbsp;
&nbsp;
&nbsp;    /* -------------------------  EDIÇÃO DE CATEGORIAS  ------------------------- */
&nbsp;
&nbsp;    @Test
&nbsp;    public void editarCategoriaComSucesso() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.findById(any())).thenReturn(Optional.ofNullable(categoria));</b>
<b class="fc">&nbsp;        when(categoriaRepository.save(any())).thenReturn(categoria);</b>
&nbsp;
<b class="fc">&nbsp;        ResponseCategoria response = categoriaController.editar(1, categoria);</b>
&nbsp;
<b class="fc">&nbsp;        assertTrue(categoria.getNome().equals(response.getNome()));</b>
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void editarCategoriaComExcecaoDadosConflitantes() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.existsByNomeIgnoreCase(any())).thenReturn(true);</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.editar(1, categoria);</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(409, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;nome_existente&quot;, e.getReason());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void editarCategoriaComExcecaoDadosInvalidos() throws Exception {
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.editar(1, null);</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(400, e.getRawStatusCode());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void editarCategoriaComExcecaoDataViolation() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.findById(anyInt())).thenReturn(optionalCategoriaMock);</b>
<b class="fc">&nbsp;        when(optionalCategoriaMock.get()).thenReturn(categoria);</b>
<b class="fc">&nbsp;        when(categoriaRepository.save(any())).thenThrow(new DataIntegrityViolationException(&quot;erro_insercao&quot;));</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.editar(1, categoria);</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(500, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;erro_insercao&quot;, e.getReason());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void editarCategoriaComExcecaoIllegalArgument() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.findById(anyInt())).thenReturn(optionalCategoriaMock);</b>
<b class="fc">&nbsp;        when(optionalCategoriaMock.get()).thenReturn(categoria);</b>
<b class="fc">&nbsp;        when(categoriaRepository.save(any())).thenThrow(new IllegalArgumentException(&quot;erro_inesperado&quot;));</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.editar(1, categoria);</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(500, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;erro_inesperado&quot;, e.getReason());</b>
<b class="fc">&nbsp;            assertTrue(e.getCause().toString().contains(&quot;java.lang.IllegalArgumentException&quot;));</b>
<b class="nc">&nbsp;        }</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void editarCategoriaComExcecaoNoSuchElement() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.findById(anyInt())).thenThrow(new NoSuchElementException());</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.editar(2023, categoria);</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(404, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;nao_encontrado&quot;, e.getReason());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void editarCategoriaComExcecao() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.findById(any())).thenReturn(optionalCategoriaMock);</b>
<b class="fc">&nbsp;        when(optionalCategoriaMock.get()).thenThrow(new NullPointerException());</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.editar(1, categoria);</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(500, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;erro_inesperado&quot;, e.getReason());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    /* -------------------------------------------------------------------------- */
&nbsp;
&nbsp;
&nbsp;
&nbsp;
&nbsp;
&nbsp;    /* ----------------------  ATUALIZAÇÃO DE CATEGORIAS  ----------------------- */
&nbsp;
&nbsp;    @Test
&nbsp;    public void atualizarCategoriaComSucesso() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.existsByNomeIgnoreCase(any())).thenReturn(false);</b>
<b class="fc">&nbsp;        when(categoriaRepository.existsById(anyInt())).thenReturn(true);</b>
<b class="fc">&nbsp;        when(categoriaRepository.save(any())).thenReturn(categoria);</b>
&nbsp;
<b class="fc">&nbsp;        ResponseCategoria response = categoriaController.atualizar(1, categoria);</b>
&nbsp;
<b class="fc">&nbsp;        assertTrue(categoria.getNome().equals(response.getNome()));</b>
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void atualizarCategoriaComExcecaoDadosConflitantes() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.existsByNomeIgnoreCase(any())).thenReturn(true);</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.atualizar(1, categoria);</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(409, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;nome_existente&quot;, e.getReason());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void atualizarCategoriaComExcecaoDadosInvalidos() throws Exception {
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.atualizar(1, categoriaInvalida);</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(400, e.getRawStatusCode());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void atualizarCategoriaComExcecaoNoSuchElement() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.existsById(anyInt())).thenReturn(false);</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.atualizar(2023, categoria);</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(404, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;nao_encontrado&quot;, e.getReason());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void atualizarCategoriaComExcecaoDataViolation() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.existsByNomeIgnoreCase(anyString())).thenReturn(false);</b>
<b class="fc">&nbsp;        when(categoriaRepository.existsById(anyInt())).thenReturn(true);</b>
<b class="fc">&nbsp;        when(categoriaRepository.save(any())).thenThrow(new DataIntegrityViolationException(&quot;erro_insercao&quot;));</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.atualizar(1, categoria);</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(500, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;erro_insercao&quot;, e.getReason());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void atualizarCategoriaComExcecaoIllegalArgument() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.existsByNomeIgnoreCase(anyString())).thenReturn(false);</b>
<b class="fc">&nbsp;        when(categoriaRepository.existsById(anyInt())).thenReturn(true);</b>
<b class="fc">&nbsp;        when(categoriaRepository.save(any())).thenThrow(new IllegalArgumentException(&quot;erro_inesperado&quot;));</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.atualizar(1, categoria);</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(500, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;erro_inesperado&quot;, e.getReason());</b>
<b class="fc">&nbsp;            assertTrue(e.getCause().toString().contains(&quot;java.lang.IllegalArgumentException&quot;));</b>
<b class="nc">&nbsp;        }</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void atualizarCategoriaComExcecao() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.existsByNomeIgnoreCase(anyString())).thenReturn(false);</b>
<b class="fc">&nbsp;        when(categoriaRepository.existsById(anyInt())).thenReturn(true);</b>
<b class="fc">&nbsp;        when(categoriaRepository.save(any())).thenThrow(new NullPointerException());</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.atualizar(1, categoria);</b>
&nbsp;
&nbsp;        }
<b class="fc">&nbsp;        catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(500, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;erro_inesperado&quot;, e.getReason());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    /* -------------------------------------------------------------------------- */
&nbsp;
&nbsp;
&nbsp;
&nbsp;
&nbsp;
&nbsp;    /* -------------------------  DELEÇÃO DE CATEGORIAS  ------------------------ */
&nbsp;
&nbsp;    @Test
&nbsp;    public void removerCategoriaComSucesso() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.existsById(anyInt())).thenReturn(true);</b>
&nbsp;
<b class="fc">&nbsp;        categoriaController.remover(1);</b>
&nbsp;        
<b class="fc">&nbsp;        verify(categoriaRepository).deleteById(1);</b>
&nbsp;
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void removerCategoriaComExcecaoNoSuchElement() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.existsById(anyInt())).thenReturn(false);</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.remover(1986);</b>
&nbsp;
<b class="fc">&nbsp;        } catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(404, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;nao_encontrado&quot;, e.getReason());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void removerCategoriaComExcecaoDataViolation() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.existsById(anyInt())).thenReturn(true);</b>
<b class="fc">&nbsp;        doThrow(new DataIntegrityViolationException(&quot;erro_remocao&quot;)).when(categoriaRepository).deleteById(anyInt());</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.remover(2023);</b>
&nbsp;
<b class="fc">&nbsp;        } catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(500, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;erro_remocao&quot;, e.getReason());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void removerCategoriaComExcecaoIllegalArgument() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.existsById(anyInt())).thenReturn(true);</b>
<b class="fc">&nbsp;        doThrow(new IllegalArgumentException(&quot;erro_inesperado&quot;)).when(categoriaRepository).deleteById(anyInt());</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.remover(1);</b>
&nbsp;
<b class="fc">&nbsp;        } catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(500, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;erro_inesperado&quot;, e.getReason());</b>
<b class="fc">&nbsp;            assertTrue(e.getCause().toString().contains(&quot;java.lang.IllegalArgumentException&quot;));</b>
<b class="nc">&nbsp;        }</b>
&nbsp;    }
&nbsp;
&nbsp;    @Test
&nbsp;    public void removerCategoriaComExcecao() throws Exception {
&nbsp;
<b class="fc">&nbsp;        when(categoriaRepository.existsById(anyInt())).thenReturn(true);</b>
<b class="fc">&nbsp;        doThrow(new NullPointerException(&quot;erro_inesperado&quot;)).when(categoriaRepository).deleteById(anyInt());</b>
&nbsp;
&nbsp;        try {
&nbsp;
<b class="fc">&nbsp;            categoriaController.remover(1);</b>
&nbsp;
<b class="fc">&nbsp;        } catch (ResponseStatusException e) {</b>
<b class="fc">&nbsp;            assertEquals(500, e.getRawStatusCode());</b>
<b class="fc">&nbsp;            assertEquals(&quot;erro_inesperado&quot;, e.getReason());</b>
<b class="nc">&nbsp;        }</b>
&nbsp;    }
&nbsp;
&nbsp;    /* -------------------------------------------------------------------------- */
&nbsp;
&nbsp;}
</code>
</pre>
</div>

<script type="text/javascript">
(function() {
    var msie = false, msie9 = false;
    /*@cc_on
      msie = true;
      @if (@_jscript_version >= 9)
        msie9 = true;
      @end
    @*/

    if (!msie || msie && msie9) {
      hljs.highlightAll()
      hljs.initLineNumbersOnLoad();
    }
})();
</script>

<div class="footer">
    
    <div style="float:right;">generated on 2023-11-25 03:05</div>
</div>
</body>
</html>
